model/src/main/kotlin/VcsInfo.kt

```diff
@@ -90,32 +90,13 @@ data class VcsInfo(
             return other
         }
 
-        var type = this.type
-        if (type == VcsType.NONE && other.type != VcsType.NONE) {
-            type = other.type
-        }
-
-        var url = this.url
-        if (url.isBlank() && other.url.isNotBlank()) {
-            url = other.url
-        }
-
-        var revision = this.revision
-        if (revision.isBlank() && other.revision.isNotBlank()) {
-            revision = other.revision
-        }
-
-        var resolvedRevision = this.resolvedRevision
-        if (resolvedRevision == null && other.resolvedRevision != null) {
-            resolvedRevision = other.resolvedRevision
-        }
-
-        var path = this.path
-        if (path.isBlank() && other.path.isNotBlank()) {
-            path = other.path
-        }
-
-        return VcsInfo(type, url, revision, resolvedRevision, path)
+        return VcsInfo(
```

Comment:
commit message: Please indicate whether this change is supposed to be non- functional as it is not obvious IMO.

Reply:
> Why did you choose to prefer this over e.g. `url.takeUnless { it.isBlank() } ?: other.url`?

Because the fields should actually never be blank, but empty at most. And instead of checking for `isEmpty()`, I found it to be more speaking to make use of the fields in `VcsInfo.EMPTY`, which I can also use for `resolvedRevision` which otherwise would deviate from the `isEmpty()` check with a `null` check. This way the merge code for all the fields is more homogenous.

Reply:
Thanks, makes sense!